### ISSUE TITLE
Make entry point optional

### ISSUE THREAD

**Issue by JonnyBurger** [remotion-dev/remotion#1391]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%24167%20Rewarded-%237E24E3.svg" />](https://issuehunt.io/r/remotion-dev/remotion/issues/1391)

**🎃 This issue is part of our Hacktoberfest campaign!**
📚 [Read more about Hacktoberfest here](https://remotion.dev/hacktoberfest)
🕛 This issue is currently assigned to @Pompette
💰 Thanks to [CodeChem](https://codechem.com) for sponsoring this issue!
⚠️ **This issue can only be taken once #1377 is finished!**

Each Remotion command starts with `npx remotion [command] [entry-point]`:

```
npx remotion preview src/index.tsx
npx remotion render src/index.tsx
npx remotion benchmark src/index.tsx
```

The goal is to make it that if you omit the entry point, Remotion will detect it itself:

Rules:
- First priority: If the entry point is specified, use it.
- Second priority: Add a new rule in the config file: `Config.Preview.setEntryPoint('src/index.tsx')` (subject to change) and check if it is set. If so, use it
- Third priority: Look for common paths like `src/index.tsx`, `src/index.ts`, `src/index.js`, `remotion/index.js`. If that is the case, use `Log.verbose` to print out the path that Remotion has decided on
- Fourth priority: Fail because entry point could not be determined.

## Acceptance criteria
- Implement logic according to rules above
- Update the CLI section of the documentation to reflect that you don't have to pass the entry point anymore and document the rules of order.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $167.00)

- [<img src='https://avatars.githubusercontent.com/u/85344006?v=4' alt='remotion-dev' width=24 height=24> remotion-dev](https://issuehunt.io/u/remotion-dev) ($167.00)

### Submitted pull Requests
- [#1448 If entry point isn't specified, try to automatically detect it](https://issuehunt.io/r/remotion-dev/remotion/pull/1448)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/remotion-dev/remotion/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by ayatkyo** [comment]
Hi @JonnyBurger, I would like to work on this issue if possible.

**Comment by JonnyBurger** [comment]
@ayatkyo Yes, however note the requirement that this issue can only be started once #1377 is done. I'll ask you to confirm again once that is the case!

**Comment by JonnyBurger** [comment]
@ayatkyo has taken another issue for now, so it is free now. Still need to wait for the other one to get merged though.

**Comment by Pompette** [comment]
Hey @JonnyBurger , if it's still available I'd like to work on this one :) !

**Comment by JonnyBurger** [comment]
@Pompette Assigned! You just have to wait for #1377 to get solved.

**Comment by Pompette** [comment]
Thanks @JonnyBurger , but I'm not sure to understand if the issue #1377 is taken by someone regarding your last comment on it. Is it currently WIP ?

**Comment by JonnyBurger** [comment]
@Pompette As of 2 dass ago, the issue is still active! We chatted on Discord about it

**Comment by JonnyBurger** [comment]
@Pompette The PR that was required for this issue to get kicked off has now been merged! 🎉  Do you want to work on this issue?

**Comment by JRavi2** [comment]
Hi @JonnyBurger, I'd like to work on this issue if it's available :)

**Comment by JonnyBurger** [comment]
@JRavi2 Giving it to you! Good luck :=
